### DISCUSSION TITLE
Welcome to deadd-notification-center Discussions!

### DISCUSSION THREAD

**Discussion by phuhl** [phuhl/linux_notification_center#153] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share your designs and configs
  * Future feature wishes and ideas
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by CyberBishop** [comment]
Hey, guys, Please How do I setup up the notification stuff, I am using Parrot OS, I have installed the software my notification is just plain with just date and time no WiFi of Bluetooth buttons under
then how do I add a shortcut that will automatically launch it

![Screenshot at 2021-06-10 21-12-38](https://user-images.githubusercontent.com/53916413/121591072-9909f000-ca28-11eb-85e0-3e786acaae67.png)

**Reply by phuhl** [reply]
Hi,

for adding the buttons have a look here: https://github.com/phuhl/linux_notification_center#configuration 
Have a look in the section [buttons]. You need to comment in the `labels` and `commands` variables and set them accordingly.

For autostarting you will have to find out how your system handels autostart and launch it from there. One common way is to use the `.xinitrc` file in your home directory to launch programs. When you found an appropriate place, just start it with `deadd-notification-center &`.

Hope it helps, cheers

**Comment by balajsra** [comment]
Hello! Love this project. I have a question about customizing colors. Which option in `deadd.css` corresponds to the colors for the buttons that appear in notifications themselves? I tried adjusting the `color` under `button`, but that didn't seem to do anything.

![2021-06-16_20-06](https://user-images.githubusercontent.com/22876640/122310536-8850f280-cede-11eb-8db1-c6e4cff08569.png)

**Reply by phuhl** [reply]
Hi and welcome!

I am not entirely sure about the buttons, but I think your should get there with `.userbutton.notification`. If that doesn't work, let me know!

**Reply by balajsra** [reply]
Thanks for the response. Unfortunately, that does not seem to be it either.

![2021-06-18_17-57](https://user-images.githubusercontent.com/22876640/122619529-a5acca80-d05e-11eb-8761-8112f7654525.png)

**Reply by balajsra** [reply]
Figured it out. It was `.userbutton`.

![image](https://user-images.githubusercontent.com/22876640/122626034-22976e80-d076-11eb-9b89-ec397d3e345d.png)

**Reply by phuhl** [reply]
Nice. Btw, what is the code for the border around the notification and the rounded corners?

**Reply by phuhl** [reply]
Probably should create an issue, .userbutton.notification should be a thing!

**Reply by balajsra** [reply]
> Nice. Btw, what is the code for the border around the notification and the rounded corners?

Here is what I did for the border w/ rounded corners. This specifies a solid green (#50fa7b) border of size 2 pixels with a 10 pixel border radius.

```css
.blurredBG.notification {
    background:  rgba(40, 42, 54, 0.8);
    border: 2pt #50fa7b solid;
    border-radius: 10px;
}
```

> Probably should create an issue, .userbutton.notification should be a thing!

Sounds good, I created https://github.com/phuhl/linux_notification_center/issues/170.

**Comment by ichirou2910** [comment]
Hi! Really love this project. The README mentions progress bar, but so far I cannot see it. Also, `deadd.css` file doesn't seem to be loaded when the noti daemon starts. Do you have any idea? Im using the `deadd-notification-center-bin` package.

![image](https://user-images.githubusercontent.com/51302893/122600527-1d4c1b00-d09a-11eb-9597-c02a2d18c718.png)

**Reply by phuhl** [reply]
Hi, I think the deadd-notification-center-bin package it too old. Try building the deadd-notification-center-git package, that is up to date with these features.

Cheers

**Reply by ichirou2910** [reply]
Somehow `deadd-notification-center-git` gave out this build error

```
--  While building package glib-0.13.8.1 (scroll up to its section to see the error) using:
      /tmp/stack-377a734b1cfe30a9/glib-0.13.8.1/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/setup/setup --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1


--  While building package cairo-0.13.5.0 (scroll up to its section to see the error) using:
      /tmp/stack-377a734b1cfe30a9/cairo-0.13.5.0/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/setup/setup --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
make: *** [Makefile:40: stack] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
error making: deadd-notification-center-git
```
I guess this is a problem with glib but I'm not familiar with haskell, do you have any idea?

**Reply by mikenrafter** [reply]
If you're on Arch Linux, then Haskell will be a real problem. When building, dynamic linking is enforced. Use `ghcup`, or [`nix`](https://nixos.org/nix) to circumvent this problem.  
Alternatively, the bin version has probably updated since this posting. Give it another go?

**Reply by ichirou2910** [reply]
Ye I tried the bin version and it works totally fine, it'd be nice if the binary is in the release tab

**Reply by phuhl** [reply]
The binary is quite out of date I think, also there is still the PR https://github.com/phuhl/linux_notification_center/pull/138 which adds lot's of cool stuff, but also has braking changes.

As I am working full time atm, I won't be able to update the binary. But if someone figures out the build issue and creates a PR for that, I'd gladly review that ;)

**Comment by mikenrafter** [comment]
https://user-images.githubusercontent.com/88250914/142709481-26ea2466-b7ca-45b6-9ac1-049e17103340.mp4

Hello, An XMonad, Nix, and Arch user here.
My deadd-notification-center looks quite appealing, and, has some unique styling attributes.
So, take a [look](https://gitlab.com/mikenrafter/my-nix-config/-/tree/main/config/dot/deadd)! :D
(the <10MB file size limit makes the video encoding sub-par, but it'll do!)

**Reply by phuhl** [reply]
Sweet, how did you do the notification merging?

You should check out PR #138 (if you can build it) it has a lot more features in terms of scripting notifications! I think, you'd love that!

**Reply by mikenrafter** [reply]
I used some [fancy CSS](https://gitlab.com/mikenrafter/my-nix-config/-/blob/main/config/dot/deadd/deadd.css#L61-L84).

My config has a problem, though, and I think it's due to GTK layering... The delete-all notification button's Z-index is lower than that of the title pieces. So, my relocated button — as seen in that video, is only very rarely clickable. I've relocated it so that the notifications don't move when the button appears.

**Comment by arkoinad** [comment]
Hi!
Great job, is there a way to add widgets?(like rss feed or stock ticker?) Something along the lines of how it is in the macos.

**Reply by phuhl** [reply]
Hi @abhixec,

no, not exactly as a widget, but you can get pretty far by crafting a notification with e.g. notify-send.py. You can add buttons, make it stay in the notification center, update, etc.. If you are very brave, you can even use this Draft: https://github.com/phuhl/linux_notification_center/pull/138 and have a lot more freedom in styling and scripting your notis.

Hope that helps, cheers

**Reply by mikenrafter** [reply]
Perhaps, using some scripting/styling techniques, you could even modify the "widget"s location, so it'd be indistinguishable from an actual widget?

**Comment by christopherbruce** [comment]
Would it be possible to get the number of items in the notification center from command line so i can display it in my status bar? or just return a json object similar to `dunstctl history`?

**Reply by phuhl** [reply]
Hey @christopherbruce,

that's currently not supported. This is the issue for it: https://github.com/phuhl/linux_notification_center/issues/34

Contributions are always welcome :) I don't have time to build that atm.

Cheers
